cfn-resources/project-invitation/cmd/resource/resource.go

```diff
@@ -34,237 +30,59 @@ var UpdateRequiredFields = []string{constants.ProjectID, constants.ID}
-	}
-	currentModel.Id = &invitation.ID
-
-	// Response
-	return handler.ProgressEvent{
-		OperationStatus: handler.Success,
-		ResourceModel:   invitationToModel(currentModel, invitation),
-	}, nil
+	return CreateOp(req, prevModel, currentModel)
 }
 
 func Read(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
-	setup()
-	_, _ = log.Debugf("Read() currentModel:%+v", currentModel)
-	// Validation
-	modelValidation := validateModel(ReadRequiredFields, currentModel)
-	if modelValidation != nil {
-		return *modelValidation, nil
-	}
-
-	// Create atlas client
-	if currentModel.Profile == nil || *currentModel.Profile == "" {
-		currentModel.Profile = aws.String(profile.DefaultProfile)
-	}
-
-	client, peErr := util.NewMongoDBClient(req, currentModel.Profile)
-	if peErr != nil {
-		return *peErr, nil
-	}
-
-	invitation, res, err := client.Projects.Invitation(context.Background(), *currentModel.ProjectId, *currentModel.Id)
-	if err != nil {
-		_, _ = log.Warnf("Read - error: %+v", err)
-
-		// if invitation already accepted
-		if res.StatusCode == 404 {
-			if alreadyAccepted, _ := validateProjectInvitationAlreadyAccepted(context.Background(), client, *currentModel.Username, *currentModel.ProjectId); alreadyAccepted {
-				return progressevents.GetFailedEventByResponse("invitation has been already accepted", res.Response), nil
-			}
-		}
-
-		return progressevents.GetFailedEventByResponse(err.Error(), res.Response), nil
-	}
-
-	// Response
-	return handler.ProgressEvent{
-		OperationStatus: handler.Success,
-		ResourceModel:   invitationToModel(currentModel, invitation),
-	}, nil
+	return ReadOp(req, prevModel, currentModel)
 }
 
 func Update(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
-	setup()
-
-	_, _ = log.Warnf("Update() currentModel:%+v", currentModel)
-
-	// Validation
-	modelValidation := validateModel(UpdateRequiredFields, currentModel)
-	if modelValidation != nil {
-		return *modelValidation, nil
-	}
-
-	// Create atlas client
-	if currentModel.Profile == nil || *currentModel.Profile == "" {
-		currentModel.Profile = aws.String(profile.DefaultProfile)
-	}
-
-	client, peErr := util.NewMongoDBClient(req, currentModel.Profile)
-	if peErr != nil {
-		return *peErr, nil
-	}
-
-	invitationReq := &mongodbatlas.Invitation{
-		Roles: currentModel.Roles,
-	}
-
-	invitation, res, err := client.Projects.UpdateInvitationByID(context.Background(), *currentModel.ProjectId, *currentModel.Id, invitationReq)
-	if err != nil {
-		_, _ = log.Warnf("Update - error: %+v", err)
-		return progressevents.GetFailedEventByResponse(err.Error(), res.Response), nil
-	}
-	_, _ = log.Debugf("%s invitation updated", *currentModel.Id)
-
-	// Response
-	return handler.ProgressEvent{
-		OperationStatus: handler.Success,
-		ResourceModel:   invitationToModel(currentModel, invitation),
-	}, nil
+	return UpdateOp(req, prevModel, currentModel)
```

Comment:
one small issue with this refactor extracting methods is that it's hard to follow the changes as they're moved to other files

Reply:
I mean you can't compare easily old and new code

Reply:
yes, this is expected. All the follow-up PRs related to this resource won't have the same issue